chromium_src/chrome/installer/setup/install_worker.cc

```diff
@@ -35,69 +35,65 @@
 
 #if BUILDFLAG(ENABLE_BRAVE_VPN)
 #include "brave/components/brave_vpn/browser/connection/ikev2/win/brave_vpn_helper/brave_vpn_helper_constants.h"
-#include "brave/components/brave_vpn/browser/connection/ikev2/win/brave_vpn_helper/brave_vpn_helper_state.h"
+#include "brave/components/brave_vpn/browser/connection/ikev2/win/brave_vpn_helper/service_details.h"
 #include "brave/components/brave_vpn/common/wireguard/win/service_constants.h"
 #include "brave/components/brave_vpn/common/wireguard/win/service_details.h"
 
 namespace {
 
-// All BraveVpn services have support of "install" switch to make registration.
-constexpr char kBraveVpnServiceInstall[] = "install";
-
 base::FilePath GetBraveVPNHelperPath(const base::FilePath& target_path,
                                      const base::Version& version) {
   return target_path.AppendASCII(version.GetString())
       .Append(brave_vpn::kBraveVPNHelperExecutable);
 }
 
-bool InstallBraveVPNService(const base::FilePath& exe_path,
-                            const CallbackWorkItem&) {
+// delete `BraveVpnWireguardService` from services and remove the tray icon.
```

Comment:
TODO(bsclifton): these need to be wrapped with a guard. Maybe stored in `Local State` - but basically only execute once and after executing set this to true. Next upgrade, that would be checked and skipped as one-time removal happened already.

Reply:
Addressed by using registry; needs a unit test.